geograypher/meshes/meshes.py

```diff
@@ -162,19 +164,57 @@ def load_mesh(
             simplify_tol_meters=ROI_simplify_tol_meters,
         )
 
-        # Downsample mesh if needed
+        # Downsample mesh if needed transfer textures from original mesh to downsampled mesh
         if downsample_target != 1.0:
             # TODO try decimate_pro and compare quality and runtime
             # TODO see if there's a way to preserve the mesh colors
             # TODO also see this decimation algorithm: https://pyvista.github.io/fast-simplification/
             self.logger.info("Downsampling the mesh")
-            self.pyvista_mesh = self.pyvista_mesh.decimate(
+            intermediate_mesh = self.pyvista_mesh.decimate(
                 target_reduction=(1 - downsample_target)
             )
+            self.pyvista_mesh = self.transfer_texture(intermediate_mesh)
         self.logger.info("Extracting faces from mesh")
         # See here for format: https://github.com/pyvista/pyvista-support/issues/96
         self.faces = self.pyvista_mesh.faces.reshape((-1, 4))[:, 1:4].copy()
 
+    def transfer_texture(self, downsampled_mesh):
+        """Transfer texture from original mesh to a downsampled version using KDTree for nearest neighbor point searches
+
+        Args:
+            downsampled_mesh (pv.PolyData): The downsampled version of the original mesh
+
+        Returns:
+            pv.PolyData: The downsampled mesh with the transferred textures
+        """
+        # Only compute KDTree if active scalars are associated with points
+        if (
+            self.kdtree is None
+            and self.pyvista_mesh.active_scalars_info.association
+            == pv.FieldAssociation.POINT
+        ):
+            # Store original mesh points in KDTree for nearest neighbor search
+            self.kdtree = KDTree(self.pyvista_mesh.points)
```

Comment:
You have the kdtree as an attribute of the mesh, but you recompute it each time. I'd choose either to just compute a local version of the kdtree, or check whether you've already computed one. Unless creating the KDTree is very expensive, I would just recompute it each time. This way, there's no risk that you're using a KDTree that's not associated with the current mesh.

Reply:
I'd be interested in hearing your thoughts though if you disagree with this logic.

Reply:
Wouldn't this method only be called when we are loading in the mesh since `transfer_textures` is called in the `load_mesh` method? And I might be wrong on this but wouldn't the `load_mesh` method only be called when we first load in a mesh or use a new one? In this case, if I just compute a local version of the kdtree there would not be any extra computations.